libbeat/outputs/codec/format/format.go

```diff
@@ -38,9 +37,5 @@ func New(fmt *fmtstr.EventFormatString) *Encoder {
 }
 
 func (e *Encoder) Encode(_ string, event *beat.Event) ([]byte, error) {
-	serializedEvent, err := e.Format.RunBytes(event)
-	if err != nil {
-		logp.Err("Fail to format event (%v): %#v", err, event)
-	}
-	return serializedEvent, err
+	return e.Format.RunBytes(event)
```

Comment:
The formatter first tries to extract the fields used in the format string from the event. And only then it applies the event to the format string. One can use default values in the format string.

On error the event will be dropped (even for filebeat). That is, we have to trade excessive logging vs. 'silently' dropping events. The error can be investigated easily. Just dropping without error might confuse users potentially thinking filebeat is super unstable, just cause the configuration is not matching the event.

I checked the outputs and all but redis do log the error as err/warning/critical (depends on output and sending guarantees). I'm ok with removing the log.Err here, but then the error must be logged in the redis output. At some point we need to collect and log potentially excessive logs in a more 'sane' manner.